FIX : Formatting of Approval Letter

Rajasimhareddybolla opened the pull request:
Fix approval letter formatting to resolve issue #9324

## Proposed Changes

- Fixes #9324 
- Added PrintPreview Component for the Approval Letter
The updated Approval Letter formatting looks like this:
<img width="1439" alt="image" src="https://github.com/user-attachments/assets/ac536a36-ee6e-41ce-9eb1-a2275b77b673">

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a `ResourcePrintPreview` component for displaying a print preview of resource details.
	- Added a new route for printing resource details, enhancing navigation options.
	- Enhanced the `ResourceDetails` component with improved layout and loading state handling.

- **Bug Fixes**
	- Improved rendering logic to ensure proper display of resource details during data fetching.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Rajasimhareddybolla committed:
Update ResourceDetails.tsx

Fix approval letter formatting to resolve issue #9324

rithviknishad reviewed (changes requested):
print preview already provides the button to print right? why duplicate again?

also let's switch to using `Button` from `ui` directory instead of `ButtonV2`.

Keep the UI identical to how it's present in other places

Rajasimhareddybolla commented:
Hi @rithviknishad, 

Thanks for your quick feedback!

1. **Duplicate Print Button**: I understand the issue of the duplicate print button due to the existing `PrintPreview` component. To resolve this, I've implemented conditional rendering to hide the redundant print button when the `PrintPreview` component is available.

2. **Button Component**: As requested, I’ve switched to using the primary variant of the `Button` component from the `ui` directory instead of `ButtonV2`.

Here’s the updated outcome:

![Updated Approval Letter Screenshot](https://github.com/user-attachments/assets/00214954-9b2d-4f76-97b5-573a067fba78)

Please let me know if any further changes are required.

Rajasimhareddybolla committed:
Add PrintPreview component with optional controls for print mode
added printPreview mode for approvalLetter

modamaan reviewed (changes requested):
![Screenshot 2024-12-07 142006](https://github.com/user-attachments/assets/738044f3-89a7-4f1e-8cbd-cec83590308e)
it is a printPreview wrapping issue. Only the `PrintPreview` component's print button is needed. Remove the `ResourceDetails` print approval letter and close button

Reference
https://github.com/ohcnetwork/care_fe/pull/8954

Rajasimhareddybolla committed:
removed duplicate print and close buttons

modamaan reviewed (changes requested):
We need to solve this without making changes to the PrintPreview reusable component

Rajasimhareddybolla commented:
Without adding the hideControls prop, the breadcrumb navigation bar appears duplicated. If it's fine, we can remove the 'Resource Details' navigation and use the 'PrintPreview' one instead. Let me know your thoughts!

modamaan commented:
@Rajasimhareddybolla  yes

Rajasimhareddybolla commented:
@modamaan I believe we can't remove the duplicate breadcrumb without adding an additional prop because the Page component is creating both the title and breadcrumb. Since the Page component is used to render both 'Resource Details' and 'Approval Letter', we might need to either add a prop to control this behavior or create an entirely new component specifically for PrintPreview to avoid duplication.
```
<Page
  title={"Resource details"}
  crumbsReplacements={{ [props.id]: { name: data.title } }}
  backUrl={"/resource/board"}
>
  {isPrintMode ? (
    <div className="my-4">
      <PrintPreview title="Approval Letter">
        {ApprovalLetter(data)}
      </PrintPreview>
    </div>
  ) : (
    // your other component here
  )}
</Page>
```
Please check this .

modamaan commented:
Don't want to remove it, try like this. 
![Screenshot_20241207_165632_Chrome](https://github.com/user-attachments/assets/63a9dbbe-98de-412a-898a-6a3975f47401)

Rajasimhareddybolla committed:
reverted changes in PrintPreview component and added the PrintView rendering

modamaan reviewed (approved):
@rithviknishad  LGTM

rithviknishad reviewed (changes requested):
Let's keep the print preview in a separate route of it's own. When hitting the back button from the breadcrumbs while in print mode, the we are going back to the resources list page instead of the Resource Details page.

Rajasimhareddybolla committed:
Add ResourcePrintPreview component and update routing for print view

Rajasimhareddybolla commented:
@rithviknishad, as you mentioned, I have created a new route for the PrintPreview such that:

The back button now navigates to the Resource Details page instead of the Resources List page.
This ensures that the PrintPreview is in a separate route of its own, aligning with the expected behavior.

nihal467 commented:
@rithviknishad can you look into it

Jacobjeevan commented on src/components/Resource/ResourceDetails.tsx:
```diff
@@ -53,321 +51,172 @@ export default function ResourceDetails(props: { id: string }) {
+    >
+      <div className="mx-3 mb-10 md:mx-8">
+        <div className="my-4 flex flex-col items-start md:flex-row md:items-center md:justify-between">
+          <ButtonV2 href={`/resource/${props.id}/print`}>
+            <CareIcon icon="l-file-alt" className="mr-2 text-lg" /> Approval
+            Letter
+          </ButtonV2>
```
Switch out the button as coderabbit suggested.

Jacobjeevan commented on src/components/Resource/ResourceDetails.tsx:
```diff
@@ -53,321 +51,172 @@ export default function ResourceDetails(props: { id: string }) {
+              <div className="pr-16 sm:px-16 sm:text-center">
+                <p className="font-bold text-primary-800">
+                  <span className="inline">
+                    Assigned to: {formatName(data.assigned_to_object)} -{" "}
```
Let's add translations for all these.

Search for existing key-value pairs in en.json, if not add them, and then use useTranslation hook.

Jacobjeevan commented on src/components/Resource/ResourceDetails.tsx:
```diff
@@ -53,321 +51,172 @@
+        <div className="mt-4 h-full rounded-lg border bg-white p-4 text-black shadow">
+          <div className="mb-4 flex flex-col sm:flex-row sm:justify-between">
+            <div className="text-xl font-semibold">{data.title || "--"}</div>
+            <ButtonV2
+              data-testid="update-status"
+              className="mt-4 sm:mt-2"
+              href={`/resource/${data.id}/update`}
+            >
+              Update Status/Details
+            </ButtonV2>
```
Switch out the button here as well.

Jacobjeevan commented on src/components/Resource/ResourcePrintPreview.tsx:
```diff
@@ -0,0 +1,166 @@
+            data.status === "ON HOLD" ? (
+              <div className="mt-4">
+                <span className="leading-relaxed">
+                  The request for resource (details below) placed by yourself is{" "}
```
Let's get translations added for all these as well.

Jacobjeevan commented on src/components/Resource/ResourcePrintPreview.tsx:
```diff
@@ -0,0 +1,166 @@
+              </span>
+              {data.reason || "--"}
+            </div>
+            <div className="mt-4">
```
Since these styles are reoccurring, let's create a common fn that renders them.

nihal467 commented:
@Rajasimhareddybolla what's the status of the PR
